### PR TITLE
Add generator stop time setting and group warm-up settings in own page

### LINKED ISSUE #748: Generator start/stop: "warm-up", "cool-down" & "generator stop time"
This feature was recently added to gui-v1 and must be added to gui-v2 as well. 

See issue: victronenergy/venus/#1171

Required changes:

1. Add new setting "Generator stop time" like: 
``` 
MbSpinBox {
	description: qsTr("Generator stop time")
	item {
		bind: Utils.path(settingsBindPrefix, "/GeneratorStopTime")
		unit: "s"
		decimals: 0
		step: 1
	}
}
```
2. Combine the settings "Warm-up time", "Cool-down time" and "Generator stop time" into a new submenu "Warm-up & cool-down" under "Generator start/stop settings". See screenshots of gui-v1:

image_480

image_480

### PR DESCRIPTION
**Pull request by jpetrell**
Fixes #748.

### CODE REVIEW

**Comment by jpetrell** [comment]
Verified. Got steps from Rein, documenting here also:

1. Add necessary capabilities:
```
mount -o remount rw / # make partition modifeable
nano /opt/victronenergy/dbus-generator-starter/startstop.py # line 353 Capabilities to 1
```

2. Check the values change:
```
dbus-spy # -> com.victronenergy.settings -> Settings/Generator[0|1] -> [CoolDownTime|GeneratorStopTime|WarmupTime]
```

File: pages/settings/PageSettingsGenerator.qml [hunk 1]
```diff
@@ -38,25 +38,51 @@ Page {
...
+									dataItem.uid: settingsBindPrefix + "/GeneratorStopTime"
+									suffix: "s"
+									decimals: 0
+									stepSize: 1
+								}
+							}
+						}
+					}
+				}
+				}
```
Comment (blammit) [inline pages/settings/PageSettingsGenerator.qml:85]: Something odd happening here with two closing braces aligned together?

File: pages/settings/PageSettingsGenerator.qml [hunk 2]
```diff
@@ -38,25 +38,51 @@ Page {
...
+				onClicked: Global.pageManager.pushPage(warmupPageComponent, { title: text })
+				Component {
+					id: warmupPageComponent
+					Page {
+						GradientListView {
+							id: settingsListView
+
+							model: ObjectModel {
+
+
```
Comment (blammit) [inline pages/settings/PageSettingsGenerator.qml:54]: Extra newlines?

File: pages/settings/PageSettingsGenerator.qml (hunk 1 above)
Comment (jpetrell) [reply pages/settings/PageSettingsGenerator.qml]: Fixed.

File: pages/settings/PageSettingsGenerator.qml (hunk 2 above)
Comment (jpetrell) [reply pages/settings/PageSettingsGenerator.qml]: Fixed.

File: pages/settings/PageSettingsGenerator.qml [hunk 3]
```diff
@@ -38,24 +38,48 @@ Page {
...
-				text: qsTrId("page_settings_generator_cool_down_time")
+			ListNavigationItem {
+				//% "Warm-up & cool-down"
+				text: qsTrId("settings_page_generator_warm_up_cool_down")
 				visible: capabilities.value & warmupCapability
-				dataItem.uid: settingsBindPrefix + "/CoolDownTime"
-				suffix: "s"
-				decimals: 0
-				stepSize: 10
+				onClicked: Global.pageManager.pushPage(warmupPageComponent, { title: text })
```
Comment (blammit) [inline pages/settings/PageSettingsGenerator.qml:45]: Could you add a newline between the onClicked and the Component below? Otherwise LGTM.

**Review by blammit — APPROVED** [review]

File: pages/settings/PageSettingsGenerator.qml (hunk 3 above)
Comment (jpetrell) [reply pages/settings/PageSettingsGenerator.qml:45]: Sure. :blush:

**Pull request merged**